src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -107,87 +108,112 @@ public static string Escape(string input)
             {
                 if (IsMetachar(input[i]))
                 {
-                    StringBuilder sb = StringBuilderCache.Acquire();
-                    char ch = input[i];
-                    int lastpos;
+                    return EscapeImpl(input, i);
+                }
+            }
 
-                    sb.Append(input, 0, i);
-                    do
-                    {
-                        sb.Append('\\');
-                        switch (ch)
-                        {
-                            case '\n':
-                                ch = 'n';
-                                break;
-                            case '\r':
-                                ch = 'r';
-                                break;
-                            case '\t':
-                                ch = 't';
-                                break;
-                            case '\f':
-                                ch = 'f';
-                                break;
-                        }
-                        sb.Append(ch);
-                        i++;
-                        lastpos = i;
+            return input;
+        }
 
-                        while (i < input.Length)
-                        {
-                            ch = input[i];
-                            if (IsMetachar(ch))
-                                break;
+        private static string EscapeImpl(string input, int i)
+        {
+            // For small inputs we allocate on the stack. In most cases a buffer three 
+            // times larger the original string should be sufficient as usually not all 
+            // characters need to be encoded.
+            // For larger string we rent the input string's length plus a fixed 
+            // conservative amount of chars from the ArrayPool.
+            Span<char> buffer = input.Length <= 80 ? stackalloc char[EscapeMaxBufferSize] : default;
```

Comment:
Using hardcoded `80` together with named constant `EscapeMaxBufferSize` does not look very great. In case somebody changes `EscapeMaxBufferSize`, this code will be likely wrong.